src/main/java/com/example/wdmsystem/employee/system/EmployeeService.java

```diff
@@ -109,6 +109,6 @@ public boolean isOwnedByCurrentUser(int employeeId) {
                 .getPrincipal();
 
         Employee employee = getEmployee(employeeId);
-        return employee.getMerchantId() == currentUser.getMerchantId();
+        return employee.getMerchant() == currentUser.getMerchant();
```

Comment:
This method should probably check ids with employee.getMerchant().getId() == currentUser.getMerchantId() as I'm not sure if comparing objects will work as intended.